src/components/Insights/InsightsPage/index.tsx

```diff
@@ -0,0 +1,523 @@
+    InsightType.EndpointHighNumberOfQueries,
+    InsightType.EndpointQueryOptimization
+  ];
+
+  return insights.findIndex((insight) =>
+    insightsWithJiraButton.includes(insight.type)
+  );
+};
+
+const renderInsightCard = (
+  insight: GenericCodeObjectInsight,
+  onJiraTicketCreate: (
+    insight: GenericCodeObjectInsight,
+    spanCodeObjectId: string | undefined,
+    event?: string
+  ) => void,
+  isJiraHintEnabled: boolean
+): JSX.Element | undefined => {
+  const handleErrorSelect = (errorId: string, insightType: InsightType) => {
+    sendTrackingEvent(globalTrackingEvents.USER_ACTION, {
+      action: `Follow ${insightType} link`
+    });
+    window.sendMessageToDigma({
+      action: actions.GO_TO_ERROR,
+      payload: {
+        errorId
+      }
+    });
+  };
+
+  const handleErrorsExpandButtonClick = () => {
+    window.sendMessageToDigma({
+      action: actions.GO_TO_ERRORS
+    });
+  };
+
+  const handleHistogramButtonClick = (
+    instrumentationLibrary: string,
+    name: string,
+    insightType: InsightType,
+    displayName: string
+  ) => {
+    window.sendMessageToDigma({
+      action: actions.OPEN_HISTOGRAM,
+      payload: {
+        instrumentationLibrary,
+        name,
+        insightType,
+        displayName
+      }
+    });
+  };
+
+  const handleLiveButtonClick = (prefixedCodeObjectId: string) => {
+    window.sendMessageToDigma({
+      action: actions.OPEN_LIVE_VIEW,
+      payload: {
+        prefixedCodeObjectId
+      }
+    });
+  };
+
+  const handleTraceButtonClick = (
+    trace: Trace,
+    insightType: InsightType,
+    spanCodeObjectId?: string
+  ) => {
+    window.sendMessageToDigma({
+      action: actions.GO_TO_TRACE,
+      payload: {
+        trace,
+        insightType,
+        spanCodeObjectId
+      }
+    });
+  };
+
+  const handleCompareButtonClick = (
+    traces: [Trace, Trace],
+    insightType: InsightType
+  ) => {
+    window.sendMessageToDigma({
+      action: actions.GO_TO_TRACE_COMPARISON,
+      payload: {
+        traces,
+        insightType
+      }
+    });
+  };
+
+  const handleAssetLinkClick = (
+    spanCodeObjectId: string,
+    insightType: InsightType
+  ) => {
+    sendTrackingEvent(globalTrackingEvents.USER_ACTION, {
+      action: `Follow ${insightType} link`
+    });
+    window.sendMessageToDigma({
+      action: actions.GO_TO_ASSET,
+      payload: {
+        spanCodeObjectId
+      }
+    });
+  };
+
+  const handleRecalculate = (
+    prefixedCodeObjectId: string,
+    insightType: InsightType
+  ) => {
+    window.sendMessageToDigma({
+      action: actions.RECALCULATE,
+      payload: {
+        prefixedCodeObjectId,
+        insightType
+      }
+    });
+  };
+
+  const handleRefresh = (insightType: InsightType) => {
+    window.sendMessageToDigma({
+      action: actions.REFRESH_ALL,
+      payload: {
+        insightType
+      }
+    });
+  };
+
+  if (isSpanDurationsInsight(insight)) {
+    return (
+      <DurationInsight
```

Comment:
Please check dev console for React key warnings when there are multiple insights with the same type
If there are errors, please use the complex key
`${insightType}-${spanCodeObjectId}`